### ISSUE TITLE
setEditorState is not a function

### ISSUE THREAD

**Issue by ruoshui-git** [ngs/draft-js-markdown-shortcuts-plugin#61]
I'm using these versions:
    "react": "^16.9.0",
    "react-dom": "^16.9.0",
    "react-scripts": "3.1.1",
    "draft-js": "^0.11.0",
    "draft-js-markdown-shortcuts-plugin": "^0.5.0",
    "draft-js-plugins-editor": "^3.0.0"

and whenever I try to use a markdown shortcuts, this error shows up in the console and the shortcut would not work:

```
Uncaught TypeError: setEditorState is not a function
    at Object.handleBeforeInput (index.js:221)
    at index.js:348
    at Array.some (<anonymous>)
    at Object.handleBeforeInput (index.js:347)
    at editOnBeforeInput (editOnBeforeInput.js:83)
    at DraftEditor.react.js:486
    at unstable_runWithPriority (scheduler.development.js:674)
    at runWithPriority$2 (react-dom.development.js:11834)
    at flushControlled (react-dom.development.js:22993)
    at DraftEditor.react.js:485
    at HTMLUnknownElement.callCallback (react-dom.development.js:363)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:412)
    at invokeGuardedCallback (react-dom.development.js:466)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:481)
    at executeDispatch (react-dom.development.js:614)
    at executeDispatchesInOrder (react-dom.development.js:639)
    at executeDispatchesAndRelease (react-dom.development.js:744)
    at executeDispatchesAndReleaseTopLevel (react-dom.development.js:753)
    at Array.forEach (<anonymous>)
    at forEachAccumulated (react-dom.development.js:723)
    at runEventsInBatch (react-dom.development.js:770)
    at runExtractedPluginEventsInBatch (react-dom.development.js:916)
    at handleTopLevel (react-dom.development.js:6171)
    at batchedEventUpdates (react-dom.development.js:2422)
    at dispatchEventForPluginEventSystem (react-dom.development.js:6271)
    at dispatchEvent (react-dom.development.js:6301)
    at unstable_runWithPriority (scheduler.development.js:674)
    at runWithPriority$2 (react-dom.development.js:11834)
    at discreteUpdates$1 (react-dom.development.js:22935)
    at discreteUpdates (react-dom.development.js:2440)
    at dispatchDiscreteEvent (react-dom.development.js:6254)
```

Is anyone else having this problem?

**Comment by davidwieler** [comment]
Yeah, having this exact same problem.

I'll comment back if I find a solution.

@ruoshui-git are you using hooks per chance?

**Comment by ruoshui-git** [comment]
> ...
> @ruoshui-git are you using hooks per chance?
Yes I am using hooks.

**Comment by philohelp** [comment]
I have the same error with the autolist and the shortcut plugin. I guess it is related to the last version of react b ut I dont know exactly how.

**Comment by davidwieler** [comment]
I couldn't figure out the issue, and actually dropped DraftJS from my project, switched to SlateJS.

**Comment by ruoshui-git** [comment]
> So, no one knows?

At least I haven't figured it out yet. And I won't be able to work on this for a very long time (like a year).

**Comment by spoeken** [comment]
It looks like it "just" needs to be refactored to work with hooks.

**Comment by mbecker-bemit** [comment]
What Versions are you using?
As I'm using React 16.10, it seem I also need `draft-js-plugins-editor:3.0.0` and `draft-js:0.11.2`

In the plugins editor the `handleBeforeInput` [changed](https://github.com/draft-js-plugins/draft-js-plugins/blob/93f9d32b703cf8e65e3576a5803388b362d19ba5/draft-js-plugins-editor/src/index.d.ts#L65), `eventTimeStamp` was added.

I opened a pull request #62  that should fix that.

**Comment by tuzmusic** [comment]
I'm not using hooks and I'm having this problem. I also tried the sample code in the Readme and had the same error.
    "draft-js": "^0.11.4",
    "react": "^16.13.1",
    "draft-js-markdown-shortcuts-plugin": "^0.5.0",
    "draft-js-plugins-editor": "^3.0.0",

**Comment by jasuperior** [comment]
This version doesnt seem to be the one which is on npm 🤦

**Comment by elbakerino** [comment]
Still not merged/fixed in the official one, PR is open, in the meantime i published it as [draft-js-md-keyboard-plugin](https://www.npmjs.com/package/draft-js-md-keyboard-plugin)

**Comment by ngs** [comment]
Published v0.6.0 that supports draft-js v0.11.x

**Comment by bw** [comment]
Hey @ngs — I am still getting this error (on a next.js app). It's unfortunately very sporadic and difficult to reproduce, but when it does happen, I get a `setEditorState is not a function` .

I'm on

- draft-js 0.11.6
- draft-js-markdown-shortcuts-plugin 0.6.1
- draft-js-plugins-editor 3.0.0
- markdown-draft-js 2.2.1
